### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by querqueq** [elastic/package-registry#955]
I'm running the whole Elastic Stack on Windows in an air-gapped environment. Which wasn't a problem yet but now I want to utilize Elastic Agent as much as possible since it really cuts down on overhead and maintenance. I tried compiling EPR on Windows which did work. But when I run it an error occurs:

```
2023-02-06T13:18:55.123+0100    INFO    package-registry/main.go:304    Packages paths: ./package-storage
2023-02-06T13:18:55.123+0100    INFO    package-registry/main.go:305    Cache time for /: 10s
2023-02-06T13:18:55.123+0100    INFO    package-registry/main.go:306    Cache time for /index.json: 10s
2023-02-06T13:18:55.123+0100    INFO    package-registry/main.go:307    Cache time for /search: 10m0s
2023-02-06T13:18:55.124+0100    INFO    package-registry/main.go:308    Cache time for /categories: 10m0s
2023-02-06T13:18:55.124+0100    INFO    package-registry/main.go:309    Cache time for all others: 10m0s
2023-02-06T13:18:55.124+0100    INFO    package-registry/main.go:140    Package registry started
2023-02-06T13:18:55.249+0100    INFO    packages/packages.go:227        Searching packages in ./package-storage
2023-02-06T13:18:55.251+0100    FATAL   package-registry/main.go:315    Init failed     {"error": "reading packages from filesystem failed: loading package failed (path: package-storage\\activemq-0.6.0.zip): open activemq-0.6.0\\manifest.yml: file does not exist", "errorVerbose": "open activemq-0.6.0\\manifest.yml: file does not exist\nloading package failed (path: package-storage\\activemq-0.6.0.zip)\ngithub.com/elastic/package-registry/packages.(*FileSystemIndexer).getPackagesFromFileSystem\n\tC:/package-registry/packages/packages.go:231\ngithub.com/elastic/package-registry/packages.(*FileSystemIndexer).Init\n\tC:/package-registry/packages/packages.go:182\nmain.CombinedIndexer.Init\n\tC:/package-registry/indexer.go:29\nmain.ensurePackagesAvailable\n\tC:/package-registry/main.go:313\nmain.initIndexer\n\tC:/package-registry/main.go:229\nmain.initServer\n\tC:/package-registry/main.go:239\nmain.main\n\tC:/package-registry/main.go:145\nruntime.main\n\tC:/Users/querqueq/.g/go/src/runtime/proc.go:250\nruntime.goexit\n\tC:/Users/querqueq/.g/go/src/runtime/asm_amd64.s:1594\nreading packages from filesystem failed\ngithub.com/elastic/package-registry/packages.(*FileSystemIndexer).Init\n\tC:/package-registry/packages/packages.go:184\nmain.CombinedIndexer.Init\n\tC:/package-registry/indexer.go:29\nmain.ensurePackagesAvailable\n\tC:/package-registry/main.go:313\nmain.initIndexer\n\tC:/package-registry/main.go:229\nmain.initServer\n\tC:/package-registry/main.go:239\nmain.main\n\tC:/package-registry/main.go:145\nruntime.main\n\tC:/Users/querqueq/.g/go/src/runtime/proc.go:250\nruntime.goexit\n\tC:/Users/querqueq/.g/go/src/runtime/asm_amd64.s:1594"}
```

Sadly I'm not familiar with golang at all. I'd appreciate any pointers on how to get this to run or if my approach is generally futile.

Here's the formatted error:

```
{
    "error": "reading packages from filesystem failed: loading package failed (path: package-storage\\activemq-0.6.0.zip): open activemq-0.6.0\\manifest.yml: file does not exist",
    "errorVerbose": "open activemq-0.6.0\\manifest.yml: file does not exist
    loading package failed (path: package-storage\\activemq-0.6.0.zip)
    github.com/elastic/package-registry/packages.(*FileSystemIndexer).getPackagesFromFileSystem
    	C:/package-registry/packages/packages.go:231
    github.com/elastic/package-registry/packages.(*FileSystemIndexer).Init
    	C:/package-registry/packages/packages.go:182
    main.CombinedIndexer.Init
    	C:/package-registry/indexer.go:29
    main.ensurePackagesAvailable
    	C:/package-registry/main.go:313
    main.initIndexer
    	C:/package-registry/main.go:229
    main.initServer
    	C:/package-registry/main.go:239
    main.main
    	C:/package-registry/main.go:145
    runtime.main
    	C:/Users/querqueq/.g/go/src/runtime/proc.go:250
    runtime.goexit
    	C:/Users/querqueq/.g/go/src/runtime/asm_amd64.s:1594
    reading packages from filesystem failed
    github.com/elastic/package-registry/packages.(*FileSystemIndexer).Init
    	C:/package-registry/packages/packages.go:184
    main.CombinedIndexer.Init
    	C:/package-registry/indexer.go:29
    main.ensurePackagesAvailable
    	C:/package-registry/main.go:313
    main.initIndexer
    	C:/package-registry/main.go:229
    main.initServer
    	C:/package-registry/main.go:239
    main.main
    	C:/package-registry/main.go:145
    runtime.main
    	C:/Users/querqueq/.g/go/src/runtime/proc.go:250
    runtime.goexit
    	C:/Users/querqueq/.g/go/src/runtime/asm_amd64.s:1594"
}
```

I copied the packages from the docker distribution image.

**Comment by jsoriano** [comment]
Hey @querqueq, thanks for reporting this!

Looks like some misuse of the filesystem abstraction in Go, that we have overlooked because we are not running tests for windows. I think it'd be nice to support this. I will keep the issue open and hope we can prioritize this soon.

**Comment by jsoriano** [comment]
Yes, I have taken a quick look to the code and it seems we are using `filepath` for paths in an abstracted filesystem (the zip reader), this is probably wrong for windows.

The task here would be to:
* Run tests on windows in CI.
* Review uses of `filepath` in `packages/fs.go`. In the zip reader we should be likely using `path` instead of `filepath`.

**Comment by querqueq** [comment]
I did what you suggested and it works now. But I also had to change `NewDataStream` in datastream.go and `NewPackage` in package.go since there are specific checks for the `manifest.yml` and `README.md`. Those paths have also been created with `filepath`. Switching to `path` fixed it but I feel like if there's gonna be another filesystem for packages (e.g. just an extracted folder) it will break again on Windows.
